### PR TITLE
Feature :: Image widget enhancements 

### LINKED ISSUE #2994: [ Enhancement ] ::  image widget enhancements
### Is your proposal related to a problem?

1) Zoom in zoom out 
2) Fit to content , fit to boundary , stretch options addition
3) Loading state 

### Describe the solution you'd like

Add requested features.

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by stepinfwd**
resolves #2994

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -78,6 +78,7 @@
     "react-table": "^7.6.3",
     "react-table-plugins": "^1.3.1",
     "react-tooltip": "^4.2.18",
+    "react-zoom-pan-pinch": "^2.1.3",
```
Comment (Navaneeth-pk) [inline frontend/package.json:81]: Changes to `frontend/package-lock.json` file also needs to be committed.

Comment (sherfin94) [reply frontend/package.json:81]: Not sure if this is related to this, but `npm install` is failing for me:
image
@stepinfwd

**Comment by sherfin94** [comment]
Seems to work well!

But I couldn't understand what `Object fit` is? What are the other options for it? Can we use a dropdown for it instead of code hinter?

image

Please see if we an apply a more unambiguous name for it, and please also document it.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Requested some changes.

Plus one other major issue: Try creating an image widget and set some custom image from google images on develop branch. Then switch to this branch, image's `fit` will change and it will appear in a different manner. We need to display the existing images exactly as they are.

Also, the buttons are outside the widget right now, a better way would be to display them in the left bottom or right bottom, smaller buttons, transparent, and only make them fully visible when mouse is hovered over the image.

**Comment by stepinfwd** [comment]
> Seems to work well!
> 
> But I couldn't understand what `Object fit` is? What are the other options for it? Can we use a dropdown for it instead of code hinter?
> 
> image
> 
> Please see if we an apply a more unambiguous name for it, and please also document it.

We will change it to drop down object fit has : cover , contain , auto etc as values same as css objectFit

**Comment by stepinfwd** [comment]
Updated documentation @sherfin94

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**